Pyspark/tests/test_regular_patients.py

```diff
@@ -0,0 +1,201 @@
+
+@pytest.fixture
+def filtered_data(spark):
+    '''Test dataframe data'''
+    data = [
+        (1, '2016-04-01'),
+        (1, '2016-09-01'),
+        (2, '2016-09-01'),
+        (3, '2016-09-01'),
+        (4, '2016-09-01'),
+        (1, '2016-08-01'),
+        (3, '2016-08-01'),
+        (4, '2016-08-01'),
+        (1, '2016-07-01'),
+        (1, '2016-06-01'),
+        (1, '2016-05-01'),
+        (2, '2016-03-01'),
+        (2, '2016-01-01'),
+        (4, '2016-01-01')
+    ]
+    schema = StructType([
+                        StructField('patient_id', StringType()),
+                        StructField('effective_from_date_formatted', StringType())
+                         ])
+    df = spark.createDataFrame(data, schema)
+    df =  df.withColumn('effective_from_date_formatted',
+                        to_date(col('effective_from_date_formatted')))
+    return df
+
+@pytest.fixture
+def consecutive_data(spark):
+    '''Test dataframe data'''
+    data = [
+        (1, '2016-04-01', 0, 5),
+        (1, '2016-05-01', 1, 5),
+        (1, '2016-06-01', 2, 5),
+        (1, '2016-07-01', 3, 5),
+        (1, '2016-08-01', 4, 5),
+        (1, '2016-09-01', 5, 5),
+        (2, '2016-01-01', 0, 0),
+        (2, '2016-03-01', 0, 0),
+        (2, '2016-09-01', 0, 0),
+        (3, '2016-08-01', 0, 1),
+        (3, '2016-09-01', 1, 1),
+        (4, '2016-01-01', 0, 1),
+        (4, '2016-08-01', 0, 1),
+        (4, '2016-09-01', 1, 1)
+    ]
+    schema = StructType([
+        StructField('patient_id', StringType()),
+        StructField('effective_from_date_formatted', StringType()),
+        StructField('count_consec', IntegerType(), True),
+        StructField('max_count_consec', IntegerType(), True)
+    ])
+    df = spark.createDataFrame(data, schema)
+    df = df.withColumn('effective_from_date_formatted',
+                       to_date(col('effective_from_date_formatted')))
+    return df
+
+@pytest.fixture
+def final_result_data(spark):
+    '''Test dataframe data'''
+    data = [
+        (1, True, False, False),
+        (2, False, False, False),
+        (3, False, False, False),
+        (4, False, False, False)
+        ]
+    schema = StructType([
+        StructField('patient_id', StringType()),
+        StructField('5months', BooleanType()),
+        StructField('9months', BooleanType()),
+        StructField('11months', BooleanType())
+        ])
+    df = spark.createDataFrame(data, schema)
+    return df
+
+
+@pytest.fixture
+def patients_data_instance(mocker, sample_data):
+    '''Fixture to create a mock instance of PatientsData.
+    Returns: PatientsData- An instance of PatientsData with the mock loaded sample data.'''
+    #mocking PatientData.load_data
+    mock_load_data = mocker.patch('regular_patients.regular_patients.PatientsData.load_data')
+    mock_load_data.return_value = sample_data
+    patients_data = PatientsData('sample-path')
+    return patients_data
+
+def test_load_data(patients_data_instance, sample_data):
```

Comment:
Usually we do not write unit test on methis thta is using only one call of some library fucntion. 
![image](https://github.com/user-attachments/assets/46a367fd-88c1-448a-bfe7-0bc9f00c0387)
There is no sense in testing that spark.read method is working correctly